examples/phase_estimator.py

```diff
@@ -0,0 +1,182 @@
+"""Creates and simulates a phase estimator circuit.
+
+=== EXAMPLE OUTPUT ===
+Estimation with 2qbits.
+Actual, Estimation (Raw binary)
+0.0000, 0.0000 (00)
+0.1000, 0.0000 (00)
+0.2000, 0.2500 (01)
+0.3000, 0.2500 (01)
+0.4000, 0.5000 (10)
+0.5000, 0.5000 (10)
+0.6000, 0.5000 (10)
+0.7000, 0.7500 (11)
+0.8000, 0.7500 (11)
+0.9000, 0.0000 (00)
+RMS Error: 0.2915
+
+Estimation with 4qbits.
+Actual, Estimation (Raw binary)
+0.0000, 0.0000 (0000)
+0.1000, 0.1250 (0010)
+0.2000, 0.1875 (0011)
+0.3000, 0.3125 (0101)
+0.4000, 0.3750 (0110)
+0.5000, 0.5000 (1000)
+0.6000, 0.6250 (1010)
+0.7000, 0.6875 (1011)
+0.8000, 0.8125 (1101)
+0.9000, 0.8750 (1110)
+RMS Error: 0.0177
+
+Estimation with 8qbits.
+Actual, Estimation (Raw binary)
+0.0000, 0.0000 (00000000)
+0.1000, 0.1016 (00011010)
+0.2000, 0.1992 (00110011)
+0.3000, 0.3008 (01001101)
+0.4000, 0.3984 (01100110)
+0.5000, 0.5000 (10000000)
+0.6000, 0.6016 (10011010)
+0.7000, 0.6992 (10110011)
+0.8000, 0.8008 (11001101)
+0.9000, 0.8984 (11100110)
+RMS Error: 0.0011
+"""
+
+
+from collections import Counter
+import numpy as np
+
+import cirq
+from cirq.ops import ControlledGate
```

Comment:
Just use `cirq.ControlledGate` and `cirq.SingleQubitMatrixGate`.

Reply:
Fixed.